mitosheet/mitosheet/public/v3/formatting.py

```diff
@@ -1,23 +1,67 @@
-from typing import Optional
+from typing import Optional, Dict, Tuple
 
 from openpyxl.styles import Font, PatternFill
 from openpyxl.styles import NamedStyle
-from openpyxl.formatting.rule import CellIsRule
+from openpyxl.formatting.rule import FormulaRule
 from openpyxl.worksheet.worksheet import Worksheet
 from pandas import DataFrame, ExcelWriter
 
 from mitosheet.excel_utils import get_column_from_column_index
 
-# Object to map the conditional formatting operators to the openpyxl operators
-CONDITIONAL_TO_OPENPYXL_OPERATOR_MAP = {
-    'greater': 'greaterThan',
-    'less': 'lessThan',
-    'number_exactly': 'equal',
-    'number_not_exactly': 'notEqual',
-    'greater_than_or_equal': 'greaterThanOrEqual',
-    'less_than_or_equal': 'lessThanOrEqual',
+# Object to map the conditional formatting operators to the excel formulas
+CONDITION_TO_COMPARISON_FORMULA: Dict[str, str] = {
+    'greater': '>',
+    'less': '<',
+    'number_exactly': '=',
+    'number_not_exactly': '<>',
+    'greater_than_or_equal': '>=',
+    'less_than_or_equal': '<='
 }
 
+SPECIAL_FORMULAS = [
+    'string_exactly',
+    'string_not_exactly',
+    'contains',
+    'string_contains_case_insensitive',
+    'string_does_not_contain',
+    'string_starts_with',
+    'string_ends_with',
+    'boolean_is_true',
+    'boolean_is_false',
+]
+
+def get_conditional_format_rule(
+    filter_condition: str,
+    fill: Optional[PatternFill],
+    font: Optional[Font],
+    filter_value: str,
+    cell_range: str
+) -> FormulaRule:
```

Comment:
Optional[FormulaRule] I guess - and we'll want to handle the None case where we get the return. I think MyPy is likely catching this.

Reply:
Oh true